### DISCUSSION TITLE
Tool calls as separate traces - How do I only trace langgraph?

### DISCUSSION THREAD

**Discussion by RamiAwar** [langchain-ai/langsmith-sdk#712] [Q&A]
Not sure where to ask this, gonna start here.

I'm getting a lot of traces from one graph execution. Not sure why it's not just one entry, it's pretty annoying.

I didn't really do anything special in my setup, only added the langsmith env vars locally.

Any ideas?

<img width="314" alt="image" src="https://github.com/langchain-ai/langsmith-sdk/assets/8523191/8f485c20-7230-4d26-aa48-e9f752dc5455">

**Comment by runirudh** [comment]
I had the same issue. The fix suggested here works, remove tracing from env and pass callbacks to invoke
https://github.com/langchain-ai/langsmith-sdk/issues/720#issuecomment-2126427734

**Reply by RamiAwar** [reply]
it does look like the same issue, thank you for pointing it out!  @runirudh 

Fix also worked for me for now.

**Comment by hinthornw** [comment]
Thanks for reporting (and for your patience)could you share:

- python version
- langgraph version
- langsmith version
- Are you running sync or async. Which method to call?
- sample code for what you're doing

We are triaging a similar issue so want to 
a) be able to help y'all
b) see if these are the same issue

**Reply by RamiAwar** [reply]
Hey @hinthornw , it does look very similar as the fix over there also worked for me.

I'm defining a langgraph graph (build graph, compile) with multiple nodes (mostly sql toolkit tools).

To be fully transparent, I started drifting away from the 'recommended' way of using Langgraph pretty quickly as I had to do some more complicated things. My code is open source actually, but might be a bit hard to read. 

Gonna share it anyways, in case you're curious. The LLM flow lives inside `backend/dataline/llm_flow/graph`, and uses tools from the `backend/dataline/llm_flow/toolkit`.
https://github.com/RamiAwar/dataline

Sorry I can't be of more help!
